e2e: Upgrade snapshot clientset and api from v1beta1 to v1

humblec opened the pull request:
snapshot.go currently make use of snapshot v1beta1 clientset and api, it has been rolled into v1 version of clientset and api .

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

humblec committed:
e2e: use snapshot v1 clientset and apis in snapshot.go

snapshot.go currently make use of snapshot v1beta1 clientset and api,
with this commit it has been rolled into v1 clientset and api.

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

Madhu-1 commented on go.mod:
```diff
@@ -14,7 +14,8 @@ require (
 	github.com/kubernetes-csi/csi-lib-utils v0.9.1
-	github.com/kubernetes-csi/external-snapshotter/v2 v2.1.1
+	github.com/kubernetes-csi/external-snapshotter/client/v4 v4.1.0
+	github.com/kubernetes-csi/external-snapshotter/v2 v2.1.5
```
Is this needed anymore?

Madhu-1 commented:
golangci-lint failure

humblec committed:
e2e: make use of snapshot v1 api in rbd_helper.go

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

humblec committed:
e2e: make use of snapshot v1 api in cephfs.go

The import name of snapshot api package has also been changed
from `vs` to `snapapi` for consistency.

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

humblec committed:
e2e: make use of snapshot v1 api in cephfs_helper.go

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

humblec committed:
e2e: make use of snapshot v1 api in utils.go

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

humblec committed:
rebase: upgrade snapshot v1beta1 api and clientset to v1 version

At present e2e snapshot code make use of snapshot v1beta1 api and client
This patch address the same and snapshot api clientset/apis are updated
to v1 version.

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

humblec commented:
/retest  ci/centos/mini-e2e-helm/k8s-1.20

humblec commented:
/retest  ci/centos/mini-e2e-helm/k8s-1.19

humblec commented:
/retest ci/centos/mini-e2e-helm/k8s-1.20

humblec commented:
/retest ci/centos/mini-e2e-helm/k8s-1.19

Madhu-1 commented on e2e/snapshot.go:
```diff
@@ -6,8 +6,8 @@ import (
-	snapapi "github.com/kubernetes-csi/external-snapshotter/v2/pkg/apis/volumesnapshot/v1beta1"
-	snapclient "github.com/kubernetes-csi/external-snapshotter/v2/pkg/client/clientset/versioned"
+	snapapi "github.com/kubernetes-csi/external-snapshotter/client/v4/apis/volumesnapshot/v1"
+	snapshotv1 "github.com/kubernetes-csi/external-snapshotter/client/v4/clientset/versioned/typed/volumesnapshot/v1"
```
Any specific reason to change the alias/ if not can we keep this as same.

humblec replied:
sure.. reverted.

nixpanic replied:
why not stick with `snapclient`?

humblec replied:
@nixpanic its `snapclient` version, where do you see different import name?

Madhu-1 commented on go.mod:
```diff
@@ -14,7 +14,7 @@ require (
 	github.com/hashicorp/vault/api v1.0.5-0.20200902155336-f9d5ce5a171a
 	github.com/kubernetes-csi/csi-lib-utils v0.9.1
-	github.com/kubernetes-csi/external-snapshotter/v2 v2.1.1
+	github.com/kubernetes-csi/external-snapshotter/client/v4 v4.1.0
```
as we have 4.1.1 https://github.com/kubernetes-csi/external-snapshotter/releases/tag/v4.1.1 can  we update this to 4.1.1?

humblec replied:
@Madhu-1 eventhough the tag and release is available, the client package resolution to `4.1.1` was not happening to me. Is it resolving/importing `4.1.1` properly at your end ?

Madhu-1 replied:
what is the error. Maybe we can check with the kube-csi team?

humblec replied:
here it is : ```go: github.com/kubernetes-csi/external-snapshotter/client/v4@v4.1.1: reading github.com/kubernetes-csi/external-snapshotter/client/client/go.mod at revision client/v4.1.1: unknown revision client/v4.1.1``` , I can check with csi team though

nixpanic replied:
I agree that it is a little strange, and ideally we should use v4.1.1.

That said there is only kubernetes-csi/external-snapshotter#515 addressed, which does not seem to affect the client package.

Madhu-1 commented on go.mod:
```diff
@@ -39,7 +39,8 @@ replace (
 	github.com/hashicorp/vault/api => github.com/hashicorp/vault/api v1.0.5-0.20200902155336-f9d5ce5a171a
 	github.com/hashicorp/vault/sdk => github.com/hashicorp/vault/sdk v0.1.14-0.20201116234512-b4d4137dfe8b
-	github.com/kubernetes-csi/external-snapshotter/v2 => github.com/kubernetes-csi/external-snapshotter/v2 v2.1.1-0.20200504125226-859696c419ff
+	github.com/kubernetes-incubator/external-storage => github.com/kubernetes-incubator/external-storage v5.5.0+incompatible
+	google.golang.org/grpc => google.golang.org/grpc v1.35.0
```
Any specific reason to add this here? this got removed in https://github.com/ceph/ceph-csi/pull/2179/files#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6L43

humblec commented:
/retest ci/centos/mini-e2e/k8s-1.19

nixpanic commented on e2e/snapshot.go:
```diff
@@ -6,8 +6,8 @@ import (
 
-	snapapi "github.com/kubernetes-csi/external-snapshotter/v2/pkg/apis/volumesnapshot/v1beta1"
-	snapclient "github.com/kubernetes-csi/external-snapshotter/v2/pkg/client/clientset/versioned"
+	snapclient "github.com/kubernetes-csi/external-snapshotter/client/v4/apis/volumesnapshot/v1"
```
you changed other places to `snapapi`, better to use the same alias everywhere?

humblec replied:
@nixpanic I dont get above comment, in e2e files , its `snapapi` everywhere. did I miss any?

nixpanic replied:
You updated to use
```
snapapi "github.com/kubernetes-csi/external-snapshotter/client/v4/apis/volumesnapshot/v1"
```
 everywhere else. Here you rename the alias `snapapi` to `snapclient`. Better use `snapapi` here as it used to be, and what you used in other locations.

humblec replied:
ok. will have the same notion here too.. changed in the last refresh

humblec commented:
@Madhu-1 can you please review ?

humblec commented:
/retest  ci/centos/mini-e2e-helm/k8s-1.19

humblec commented:
@Mergifyio rebase
